Xwt.Gtk/Xwt.GtkBackend/AccessibleBackend.cs

```diff
@@ -0,0 +1,154 @@
+﻿//
+// AccessibleBackend.cs
+//
+// Author:
+//       Vsevolod Kukol <sevoku@microsoft.com>
+//
+// Copyright (c) 2017 Microsoft Corporation
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+using System;
+using Xwt.Accessibility;
+using Xwt.Backends;
+
+namespace Xwt.GtkBackend
+{
+	public class AccessibleBackend : IAccessibleBackend
+	{
+		Gtk.Widget widget;
+		IAccessibleEventSink eventSink;
+		ApplicationContext context;
+
+		public AccessibleBackend ()
+		{
+		}
+
+		public void Initialize (IAccessibleEventSink eventSink)
+		{
+			this.eventSink = eventSink;
+		}
+
+		public void Initialize (IWidgetBackend parentWidget, IAccessibleEventSink eventSink)
+		{
+			var backend = parentWidget as WidgetBackend;
+			Initialize (backend?.Widget, eventSink);
+		}
+
+		public void Initialize (object parentWidget, IAccessibleEventSink eventSink)
+		{
+			this.eventSink = eventSink;
+			widget = parentWidget as Gtk.Widget;
+			if (widget == null)
+				throw new ArgumentException ("The widget is not a Gtk.Widget.", nameof (parentWidget));
+		}
+
+		public void InitializeBackend (object frontend, ApplicationContext context)
+		{
+			this.context = context;
+		}
+
+		public Rectangle Bounds {
+			get {
+				int x = 0, y = 0, w = 0, h = 0;
+				(widget.Accessible as Atk.Component)?.GetExtents (out x, out y, out w, out h, Atk.CoordType.Screen);
+				return new Rectangle (x, y, w, h);
+			}
+			set {
+				(widget.Accessible as Atk.Component)?.SetExtents ((int)value.X, (int)value.Y, (int)value.Width, (int)value.Height, Atk.CoordType.Screen);
+			}
+		}
+
+		public string Description {
+			get {
+				return widget.Accessible.Description;
+			}
+			set {
+				widget.Accessible.Description = value;
+			}
+		}
+
+		public string Label {
+			get {
+				return widget.Accessible.Name;
```

Comment:
The Name property in Atk isn't a label, but an identifier such as "MyExcellentDialog.AdditionButton"

It is called accessibilityIdentifier in Cocoa.